templates/import.php

```diff
@@ -0,0 +1,269 @@
+			</div>
+			<div class="status">
+				<span id="completed-total" class="completed">0/0</span>
+				<span id="progress-total" class="progress">0%</span>
+				<span class="dashicons dashicons-yes"></span>
+			</div>
+		</div>
+	</div>
+</div>
+
+<table id="import-log" class="widefat">
+	<thead>
+		<tr>
+			<th>Type</th>
+			<th>Message</th>
+		</tr>
+	</thead>
+	<tbody>
+	</tbody>
+</table>
+
+<?php
+$args = array(
+	'action' => 'wxr-import',
+	'id'     => $this->id,
+);
+$url = add_query_arg( urlencode_deep( $args ), admin_url( 'admin-ajax.php' ) );
+
+$script_data = array(
+	'count' => array(
+		'posts' => $data->post_count,
+		'media' => $data->media_count,
+		'users' => count( $data->users ),
+		'comments' => $data->comment_count,
+		'terms' => $data->term_count,
+	),
+	'url' => $url,
+	'strings' => array(
+		'complete' => __( 'Import complete!', 'wordpress-importer' ),
+	),
+);
+?>
+<script>
+var wxrImport = {
+	complete: {
+		posts: 0,
+		media: 0,
+		users: 0,
+		comments: 0,
+		terms: 0,
+	},
+
+	updateDelta: function (type, delta) {
+		this.complete[ type ] += delta;
+
+		var self = this;
+		requestAnimationFrame(function () {
+			self.render();
+		});
+	},
+	updateProgress: function ( type, complete, total ) {
+		var text = complete + '/' + total;
+		document.getElementById( 'completed-' + type ).innerHTML = text;
+		var percent = parseInt( complete ) / parseInt( total );
+		document.getElementById( 'progress-' + type ).innerHTML = Math.round( percent * 100 ) + '%';
+		document.getElementById( 'progressbar-' + type ).value = percent * 100;
+	},
+	render: function () {
+		var types = Object.keys( this.complete );
+		var complete = 0;
+		var total = 0;
+		console.log(this);
+		for (var i = types.length - 1; i >= 0; i--) {
+			var type = types[i];
+			this.updateProgress( type, this.complete[ type ], this.data.count[ type ] );
+
+			complete += this.complete[ type ];
+			total += this.data.count[ type ];
+		}
+
+		this.updateProgress( 'total', complete, total );
+	}
+};
+wxrImport.data = <?php echo wp_json_encode( $script_data ) ?>;
+wxrImport.render();
+
+
+var evtSource = new EventSource( wxrImport.data.url );
+evtSource.onmessage = function ( message ) {
+	var data = JSON.parse( message.data );
+	switch ( data.action ) {
+		case 'updateDelta':
+			wxrImport.updateDelta( data.type, data.delta );
+			break;
+
+		case 'complete':
+			evtSource.close();
+			jQuery('#import-status-message').text( wxrImport.data.strings.complete );
+			break;
+	}
+};
+evtSource.addEventListener( 'log', function ( message ) {
+	var data = JSON.parse( message.data );
+	var row = document.createElement('tr');
+	var level = document.createElement( 'td' );
+	level.appendChild( document.createTextNode( data.level ) );
+	row.appendChild( level );
+
+	var message = document.createElement( 'td' );
+	message.appendChild( document.createTextNode( data.message ) );
+	row.appendChild( message );
+
+	jQuery('#import-log').append( row );
+});
+</script>
+<style>
```

Comment:
Any way this can go into a CSS file or head action? Not technically valid inside `<body>`